### PR TITLE
DRIVERS-2644 Downgrade unified test files to the lowest supported schema version

### PR DESCRIPTION
**Pull request by GromNaN**
Fix DRIVERS-2644
This change is necessary for https://github.com/mongodb/mongo-php-library/pull/1090#discussion_r1212163608

<details>
  <summary>Script</summary>

I made a small PHP script to detect the lowest schema version required for each test file. It can be translated to an other language and run in CI.
```php
<?php

function validateJsonFile($testFile, $schemaVersion)
{
    $schemaFile = "source/unified-test-format/schema-$schemaVersion.json";
    $command = "ajv -s $schemaFile -d $testFile";
    $output = shell_exec($command);
    return strpos($output ?? '', 'valid') !== false;
}

function modifySchemaVersion($jsonFile, $newSchemaVersion)
{
    $jsonData = file_get_contents($jsonFile);
    $jsonUpdated = preg_replace('/"schemaVersion": "(1\.[\d]+)"/', '"schemaVersion": "'.$newSchemaVersion.'"', $jsonData);
    file_put_contents($jsonFile, $jsonUpdated);

    $yamlFile = str_replace('.json', '.yml', $jsonFile);
    $yamlData = file_get_contents($yamlFile);
    $yamlUpdated = preg_replace('/schemaVersion: "(1\.[\d]+)"/', 'schemaVersion: "'.$newSchemaVersion.'"', $yamlData);
    file_put_contents($yamlFile, $yamlUpdated);
}

function getLowestSchemaVersion($jsonFile)
{
    $jsonData = file_get_contents($jsonFile);
    $data = json_decode($jsonData, true);
    if (!isset($data['schemaVersion'])) {
        return null;
    }
    $currentVersion = $data['schemaVersion'];
    for ($i = (int)substr($currentVersion, -2); $i >= 0; $i--) {
        $newVersion = "1." . $i;
        if (validateJsonFile($jsonFile, $newVersion)) {
            $currentVersion = $newVersion;
        } else {
            break;
        }
    }
    return $currentVersion;
}

$jsonFiles = [];

$iterator = new RecursiveIteratorIterator(new RecursiveDirectoryIterator('source/'));
foreach ($iterator as $file) {
    if ($file->isFile() && $file->getExtension() === 'json' && !str_contains($file->getPathname(), 'source/unified-test-format')) {
        $jsonFiles[] = $file->getPathname();
    }
}

foreach ($jsonFiles as $jsonFile) {
    $lowestVersion = getLowestSchemaVersion($jsonFile);
    if ($lowestVersion) {
        modifySchemaVersion($jsonFile, $lowestVersion);
    }
}

```
</details>

### CODE REVIEW

**Review by ShaneHarvey — APPROVED** [review]
LGTM, nice script! I'm in favor of adding a script like this to CI too.

**Comment by GromNaN** [comment]
Can we merge this PR as is and create a ticket (or use the existing ticket) to work on the CI script later?

**Comment by ShaneHarvey** [comment]
Sounds good to me.

File: source/auth/tests/unified/reauthenticate_with_retry.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "description": "reauthenticate_with_retry",
-  "schemaVersion": "1.12",
+  "schemaVersion": "1.3",
```
Comment (jmikola) [inline source/auth/tests/unified/reauthenticate_with_retry.json:3]: The `auth/tests/unified` files are missing corresponding changes to YAML files.

Note that the YAML files are canonical, so the PHP script you have to check for minimum schema version should target those and then you should regenerate JSON files using the Makefile in `source/` for this repo.

File: source/client-side-operations-timeout/tests/legacy-timeouts.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 description: "legacy timeouts continue to work if timeoutMS is not set"
 
-schemaVersion: "1.9"
+schemaVersion: "1.0"
```
Comment (jmikola) [inline source/client-side-operations-timeout/tests/legacy-timeouts.yml:3]: I've noted that some of the CSOT spec tests are generated by template, but this one does not appear to be generated. Something to keep in mind for this, though, as some other specs like CSFLE also use templates.

**Review by jmikola — COMMENTED** [review]
I think this is just missing changes to the YAML files for the unified auth spec tests.

File: source/auth/tests/unified/reauthenticate_with_retry.json (hunk 1 above)
Comment (GromNaN) [reply source/auth/tests/unified/reauthenticate_with_retry.json:3]: Good point. I don't know how it was missed by the script.

File: source/client-side-operations-timeout/tests/legacy-timeouts.yml (hunk 2 above)
Comment (GromNaN) [reply source/client-side-operations-timeout/tests/legacy-timeouts.yml:3]: Indeed, the script doesn't take this into account at all.

**Review by jmikola — APPROVED** [review]

**Comment by alcaeus** [comment]
Created DRIVERS-2649 to track adding a script like this to CI.

**Review by alcaeus — APPROVED** [review]
LGTM!

**Pull request merged**